### ISSUE TITLE
Version toggle is not shown for translations

### ISSUE THREAD

**Issue by tharders** [ampproject/amp.dev#2963]
## Problem

Currently the importer adds the version information to all imported components.
* List of all versions
* File version
* A flag that shows the file is the latest version
* A special path mapping if the file is the latest version

That is fine as long as all documentation is imported.
But when we have translated documentation the version information is not updated!

As a result the version information is not available for the current translated files:
https://amp.dev/es/documentation/components/amp-carousel-v0.1/?format=websites

## Possible solution

It would be best, if the single files would not contain more than the file version.
In that case the translated files would not have to change when a new version is available.

A possible solution might be to put each component documentation into their own folder and set the version information in the `_blueprint.yaml`

The folder structure could be as follows:
![image](https://user-images.githubusercontent.com/6458408/64859517-bed34c80-d62a-11e9-843a-3ad312d41d18.png)

The `_blueprint.yaml` could look like this:
```yaml
$path: /documentation/components/{base}.html
$localization:
  path: /{locale}/documentation/components/{base}.html
$view: /views/detail/component-detail.j2
versions:
  - '0.2'
  - '0.1'
latest_version: '0.2'
```

In order to server the latest version with the old URL, we could create a nearly empty index.md file that would use a specific template to simply render the latest version.
Links from other files to that component should then point to the index.md when no specific version is chosen.

**Comment by tharders** [comment]
With the proposed solution the latest version would be available under two links.
At least we should ensure that the canonical link of the latest version points to the index.

Also we must check how https://github.com/ampproject/amp.dev/pull/2463 fits into the picture.

Additionally we should think about the versioning in the search.
I suggest old versions should also have the version number in the title.
But this can also be addressed separately.

**Comment by tharders** [comment]
@matthiasrohmer @sebastianbenz Please share your ideas on this topic.

**Comment by tharders** [comment]
I just saw that the version attributes are actually inherited from the translated version!
So the problem why we do not see a version toggle seems to be that the version toggle is not rendered in the normal templates, but rather when importing the documents.
So we should change that.
Since all the version information is available in the document we can easily use this in the normal templates.

**Comment by tharders** [comment]
Still the proposed directory structure would have a benefits over the current structure:
* With lots of versions and lots of translated documents the list of files gets currently quite long
* Currently a translator has to remove the version information that are in the original file in order to avoid problems when new versions are released

**Comment by sebastianbenz** [comment]
Good catch @tharders! I ran also into this while looking into #2895. It gets more complicated:

1. We need a mapping from format to supported versions: 

```
{
  websites: ["0.1", "0.2"]
  emails: ["0.1"]
}
```

We need this at build time and during runtime to be able to render the sidebar content dynamically. My idea was to generate a single JSON file containing a format -> component -> supported versions mapping during build.

2. Links such as `../amp-carousel/index.md` need to point to different versions depending on the currently selected format.

**Comment by tharders** [comment]
Was fixed with #2974
